packages/camera/camera/example/lib/main.dart

```diff
@@ -651,39 +651,45 @@ class _CameraExampleHomeState extends State<CameraExampleHome>
       imageFormatGroup: ImageFormatGroup.jpeg,
     );
 
-    controller = cameraController;
```

Comment:
With this change, wouldn't two calls to `onNewCameraSelected` in rapid succession have a race where a controller would be replaced without being disposed?

Reply:
I think it is not covered in this PR yet. Let me know if the example can be updated/refactored for better architecture or for test purposes.

Reply:
Sure, anything we can do to make the example better is a win. We should try to make sure that testability changes are minimally intrusive when possible so they don't distract from understanding of the example itself.